Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -306,6 +304,33 @@ public struct PackageBuilder {
         }
     }
 
+    /// Returns true if pathComponent indicates a reserved directory.
+    public static func isReservedDirectory(pathComponent: String) -> Bool {
+        return isPackageDirectory(pathComponent: pathComponent) ||
+            isSourceDirectory(pathComponent: pathComponent) ||
+            isTestDirectory(pathComponent: pathComponent)
+    }
+
+    /// Returns true if pathComponent indicates a package directory.
+    public static func isPackageDirectory(pathComponent: String) -> Bool {
+        return pathComponent.lowercased() == "packages"
```

Comment:
@aciidb0mb3r yeah I thought about putting all of those literal in constants. let me do that.